cloud/huawei/huaweiproviderconfig/v1alpha1/types.go

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type HuaweiProviderConfig struct {
+	metav1.TypeMeta `json:",inline"`
+
+	AvailabilityZone string `json:"availability_zone"`
+	// need more element for HuaweiProviderConfig
```

Comment:
Machine name, flavor and image info are required when creating machine on huaweicloud, you can add as following:
```go
	// The name of your server instance.
	Name string `json:"name"`
	// The flavor reference, as a UUID or full URL, for the flavor for your server instance.
	FlavorRef string `json:"flavorRef"`
	// The UUID of the image to use for your server instance.
	ImageRef string `json:"imageRef"`
```